### ISSUE TITLE
Unable to click on popups in Intellij Idea (XWayland)

### ISSUE THREAD

**Issue by skerit** [labwc/labwc#360]
When a window pops up in Intellij I'm able to hover over it (so it's picking up hover actions) but when I actually click in the popup, the click "falls through" and clicks on the underlying surface instead.

This is happening for all popups: menus, context menus & advanced tooltips

It seems to be a regression, because I also tried it in 0.5.1 and it's working fine there.

**Comment by Joshua-Ashton** [comment]
This happens everywhere for me now too

**Comment by acoolstraw** [comment]
I believe you need to hold left click, go to the option you want to select and release left click. This happens to me with some application, I can't recall which one though.

**Comment by Joshua-Ashton** [comment]
Ill try and fix this today

**Comment by Consolatis** [comment]
So I assume this is about the master branch and X11 applications?

Just tried to reproduce it but failed:
- tested X11 VLC menu and context menu on empty playlist
- tested wayland native Pluma menu and context menu

Both times for directly clicking on a menu item + clicking on a menu item and then releasing the mouse on a different one.
I don't have Wine here (aarch64 system) so can't really debug that one but it should behave just like a usual X11 client.

Is there maybe some additional condition that must be met to trigger the bug?

**Comment by skerit** [comment]
Testing VLC and Chrome under XWayland works fine, indeed. Seems to only happen in Intellij Idea as far as I can tell (I don't run an awful lot of X11 applications anymore, luckily)

Could it be related to the underlying cause of issue #352 ?
One of the commits in Sway that fixed that mentioned this:

```
// Try to find another unmanaged surface from the same process to pass
// focus to. This is necessary because some applications (e.g. Jetbrains
// IDEs) represent their multi-level menus as unmanaged surfaces, and
// when closing a submenu, the main menu should get input focus.
```
https://github.com/swaywm/sway/commit/613abdda6fa978480067a35c3adc46fb414b0b5c

Something-something-unmanaged surfaces?

**Comment by Joshua-Ashton** [comment]
happens to me on every dropdown

**Comment by johanmalm** [comment]
Is the regression caused by 61675b5? Should we revert that commit until we understand what's going on?

**Comment by skerit** [comment]
@johanmalm No, I tested the commit just before that (e87aa19066e5dae6ddfe1f4c5d4537474083436a) and I got the same issue.
I tried to use git bisect to find the exact commit, but I'm unable to build most of the earlier commits.

**Comment by skerit** [comment]
Found it: the first commit that has this bug is 519375a7f8588b0acf7690f2fabde2380db51529

**Comment by Consolatis** [comment]
Good find. So I think what that means is that some mousebind which was not called anymore in your fork (by reverting that commit) interferes with the proper click coming trough to the popup.
Maybe your mousebind config is ordered slightly different than the default one, .. Does the bug get fixed when you run `labwc -C /dev/null` with latest master?

**Comment by skerit** [comment]
I tried the default `rc.xml.all` file on the current master, but that did not fix it.
My custom config isn't that big, actually. I only have some `<context name="Frame">` mousebinds configured. Still need to tweak it :monocle_face: 

Haven't tried `labwc -C /dev/null` yet, because I can't figure out how to open something inside of it while running it nested in another labwc :grimacing: 

I actually already applied the suggestions mentioned on the Sway wiki. The labwc `environment` configuration file I based my config on already contained the `_JAVA_AWT_WM_NONREPARENTING=1` value :smile: 

Changing the JDK used to run Idea doesn't really apply anymore: the default is already JDK 11. Using anything newer just breaks Idea completely. (As in: it refuses to start)

**Comment by johanmalm** [comment]
You can start a terminal like this

labwc -C /dev/null -s xterm

**Comment by skerit** [comment]
Ok, then I can confirm the issue is also there in Idea when starting it in a `labwc -C /dev/null` instance.

**Comment by johanmalm** [comment]
Would someone mind checking if this commit https://github.com/labwc/labwc/commit/96315def57ee9206dad17b5777e5b1ff6af32ffe on master fixes the issue.

**Comment by skerit** [comment]
I checked it with 96315def57ee9206dad17b5777e5b1ff6af32ffe, and it's working perfectly there. Awesome :smiley: 

I also checked the latest commit in master (2a14e8271b60d24d561afbf2d9b40fea2a0aaa3b), and that actually crashes labwc when a popup tries to open in Idea by just hovering over something. (Context menu still works though)

I tried running labwc with more verbose logging, but it didn't give me a lot of info:
```
00:00:18.168 [../src/xwayland-unmanaged.c:28] handling set_geometry
(EE) failed to read Wayland events: Connection reset by peer
fish: Job 1, 'labwc -V' terminated by signal SIGSEGV (Address boundary error)
```

Just FYI, maybe you're just working on something there ^^

**Comment by johanmalm** [comment]
Oh. Thanks for letting me know. I'd better not try to handle set_geometry until I can test it with something.
Just downloading intellij now

**Comment by johanmalm** [comment]
@skerit I'm running `idea` now, but can't get it to crash. Popups when hovering over things work fine. Have tried menus/settings/hovering-over-underlined code, etc. Grateful for help.

**Comment by skerit** [comment]
Oh, strange. I managed to crash it in a regular and a nested labwc instance.
I'll recompile it and try again. Might have to do with the extra custom properties I've set:

```
suppress.focus.stealing.linux=true
suppress.focus.stealing.active.window.checks=true
suppress.focus.stealing.disable.auto.request.focus=true
```

I'll do it without those and see if it still crashes.

**Comment by skerit** [comment]
Tried it again, with & without the custom properties and it's still crashing with commit 2a14e8271b60d24d561afbf2d9b40fea2a0aaa3b. But that doesn't really matter, because there are no more crashes with the latest commit in master (71f8aca66be5d90b6a27c3afce20b5cd5bd73e20) :partying_face:

**Comment by Consolatis** [comment]
Nice. I somewhat lost track of the issues raised here and don't want to prematurely close this issue.
Is everything fixed for Intellij popups regarding keyboard focus / mouse clicks in latest mater?
